locations/hours.py

```diff
@@ -942,14 +942,19 @@ def add_range(self, day, open_time, close_time, time_format="%H:%M", closed=CLOS
         if isinstance(close_time, str):
             if close_time.lower() in closed:
                 return
-            if close_time == "24:00" or close_time == "00:00":
+            if close_time in ("24:00", "00:00", "0:00"):
                 close_time = "23:59"
-            if close_time == "24:00:00" or close_time == "00:00:00":
+            if close_time in ("24:00:00", "00:00:00"):
                 close_time = "23:59:00"
```

Comment:
On second check, `OpeningHours.as_opening_hours()` only does a midnight check on `23:59` not `23:59:00`. Perhaps this line should be `close_time = "23:59:59"` and this added to the midnight check in  `OpeningHours.as_opening_hours()`? That way the code is accommodating both precisions of time--to the minute and to the second.

Reply:
I will start from adding this case to tests.

Reply:
Note that 

```
    o = OpeningHours()
    o.add_range("Mo", "11:00:00", "24:00:00", time_format="%H:%M:%S")

    assert o.as_opening_hours() == "Mo 11:00-24:00"
```

and

```
    o = OpeningHours()
    o.add_range("Mo", "11:00:00", "00:00:00", time_format="%H:%M:%S")

    assert o.as_opening_hours() == "Mo 11:00-24:00"
```

are passing. Not relooked into code yet.

Reply:
oh, I see - that is because second part of the hack does `time.strftime("%H:%M", h[1]).replace("23:59", "24:00")`


which BTW seems to changing also say `23:59:30` into `24:00`

Reply:
see

https://github.com/alltheplaces/alltheplaces/blob/4fb79648ae39fc26261e7778fa4e6bfdbf64f0df/locations/hours.py#L997

in general the entire `struct_time` looks poorly fitting here as it cannot hold `24:00` while having say year field